openedx_filters_samples/pipeline.py

```diff
@@ -0,0 +1,838 @@
+            custom_template (str): The custom template to render.
+        """
+        log.info("Rendering custom response for course %s", context["course_id"])
+        response = HttpResponse("Here's the text of the web page.")
+        raise CertificateRenderStarted.RenderCustomResponse(
+            "You can't generate a certificate from this site.",
+            response=response,
+        )
+
+
+class RedirectToCustomCertificate(PipelineStep):
+    """
+    Redirect to custom certificate page.
+
+    By raising RedirectToPage exception, the certificate generation process will be stopped and the user will be
+    redirected to a custom certificate page or any other URL.
+
+    Example usage:
+
+    >>> OPEN_EDX_FILTERS_CONFIG = {
+            "org.openedx.learning.certificate.render.started.v1": {
+                "fail_silently": False,
+                "pipeline": [
+                    "openedx_filters_samples.pipeline.RenderCustomCertificateStep"
+                ]
+            }
+        }
+    """
+
+    def run_filter(self, context, custom_template):
+        """
+        Raise RedirectToPage exception to redirect the user to a custom certificate page.
+
+        Arguments:
+            context (dict): The context data for the certificate.
+            custom_template (str): The custom template to render.
+        """
+        log.info(
+            "Redirecting to custom certificate page for course %s", context["course_id"]
+        )
+        raise CertificateRenderStarted.RedirectToPage(
+            "You can't generate a certificate from this site, redirecting to the correct location.",
+            redirect_to="https://certificate.pdf",
+        )
+
+
+class RenderResponseCourseAbout(PipelineStep):
+    """
+    Alter the course about render process by rendering a custom response.
+
+    By raising RenderCustomResponse exception, the course about render process will be stopped and a custom response
+    will be rendered instead of the course about page. In this case, the response will be a simple HttpResponse.
+
+    Example usage:
+
+    >>> OPEN_EDX_FILTERS_CONFIG = {
+            "org.openedx.learning.course_about.render.started.v1": {
+                "fail_silently": False,
+                "pipeline": [
+                    "openedx_filters_samples.pipeline.RenderResponseCourseAbout"
+                ]
+            }
+        }
+    """
+
+    def run_filter(self, context, custom_template, *args, **kwargs):
+        """
+        Raise RenderCustomResponse exception to alter the course about render process.
+
+        When raising the exception, this filter uses a redirect_to field handled by
+        the course about view that redirects to the URL indicated.
+
+        Arguments:
+            context (dict): The context data for the course about page.
+            custom_template (str): The custom template to render.
+        """
+        log.info("Rendering custom response for course %s", context["course_id"])
```

Comment:
The `course_id` key does not exist.
```suggestion
        log.info("Rendering custom response for course %s", context["course"].id)
```

Reply:
See: https://github.com/eduNEXT/openedx-filters-samples/pull/2#discussion_r1935374013